src/Microsoft.Extensions.SecretManager.Tools/Internal/DotNetCoreSdkResolver.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+
+namespace Microsoft.Extensions.ProjectModel.Internal
+{
+    internal class DotNetCoreSdkResolver
+    {
+        private readonly string _installationPath;
+
+        public DotNetCoreSdkResolver(string installationDir)
+        {
+            _installationPath = installationDir;
+        }
+
+        private IEnumerable<string> Installed
+            => Directory.EnumerateDirectories(Path.Combine(_installationPath, "sdk"));
+
+        private string ResolveGlobalJsonSdkVersion(string start)
+        {
+            var dir = new DirectoryInfo(start);
+            FileInfo globalJson = null;
+            while (dir != null)
+            {
+                var candidate = Path.Combine(dir.FullName, "global.json");
+                if (File.Exists(candidate))
+                {
+                    globalJson = new FileInfo(candidate);
+                    break;
+                }
+                dir = dir.Parent;
+            }
+            if (globalJson == null)
+            {
+                return null;
+            }
+            var t = new { sdk = new { version = "" } };
+            try
+            {
+                var obj = JsonConvert.DeserializeAnonymousType(File.ReadAllText(globalJson.FullName), t);
+                return obj?.sdk?.version;
+            }
+            catch (JsonException)
+            {
+                // TODO log
+                return null;
+            }
+        }
+
+        public DotNetCoreSdk ResolveLatest()
+        {
+            var first = Installed.Select(d => new { path = d, version = SemanticVersion.Parse(Path.GetFileName(d)) })
+                .OrderByDescending(sdk => sdk.version)
+                .First();
+
+            return new DotNetCoreSdk
+            {
+                BasePath = first.path,
+                Version = first.version.ToFullString()
+            };
+        }
+
+        public DotNetCoreSdk ResolveProjectSdk(string projectDir)
+        {
+            var global = ResolveGlobalJsonSdkVersion(projectDir);
+            if (!string.IsNullOrEmpty(global))
+            {
+                var sdk = Installed
+                    .Where(p => Path.GetFileName(p).Equals(global, StringComparison.OrdinalIgnoreCase))
+                    .Select(d => new { path = d, version = SemanticVersion.Parse(Path.GetFileName(d)) })
+                    .First();
+
+                return new DotNetCoreSdk
+                {
+                    BasePath = sdk.path,
+                    Version = sdk.version.ToFullString()
+                };
+            }
+
+            return ResolveLatest();
```

Comment:
Odd that above you're newing up a `DotNetCoreSDK` in the case that you can resolve a globaljson sdk but not in this case. It's inconsistent. 

If you're going to have the ResolveLatest/ResolveGlobalJsonSdkVersion methods separate you should normalize their output.